go-controller/pkg/factory/factory.go

```diff
@@ -56,6 +79,97 @@ func (i *informer) forEachHandler(obj interface{}, f func(h *Handler)) {
 	}
 }
 
+func (i *informer) processEvents(eq *eventQueue, namespacedName ktypes.NamespacedName) {
+	for {
+		select {
+		case event, ok := <-eq.queue:
+			if !ok {
+				return
+			}
+
+			switch event.kind {
+			case addEvent:
+				i.runAddHandlers(event.obj)
+			case updateEvent:
+				i.runUpdateHandlers(event.oldObj, event.obj)
+			case deleteEvent:
+				i.runDeleteHandlers(event.obj)
+			}
+			// If there are no more events left in the queue,
+			// drop this object's event queue from the informer's
+			// object map
+			if func() bool {
```

Comment:
`if func`? ew
fwiw you don't even need to return here since if you don't you'll immediately return on the next iteration of the loop when you try to read from the closed channel

Reply:
Fixed.